templates/user/auth/oauth_container.tmpl

```diff
@@ -0,0 +1,21 @@
+<div class="divider divider-text">
+	{{ctx.Locale.Tr "sign_in_or"}}
+</div>
+<div id="oauth2-login-navigator" class="tw-py-1">
+	<div class="tw-flex tw-flex-col tw-justify-center">
+		<div id="oauth2-login-navigator-inner" class="tw-flex tw-flex-col tw-flex-wrap tw-items-center tw-gap-2">
+		{{range $provider := .OAuth2Providers}}
+			<a class="{{$provider.Name}} ui button tw-flex tw-items-center tw-justify-center tw-py-2 tw-w-full oauth-login-link" href="{{AppSubUrl}}/user/oauth2/{{$provider.DisplayName}}">
+				{{$provider.IconHTML 28}}
+				{{ctx.Locale.Tr "sign_in_with_provider" $provider.DisplayName}}
+			</a>
+		{{end}}
+		{{if .EnableOpenIDSignIn}}
```

Comment:
The template is shared by "signup" and "signin", but it just checks "EnableSignIn" ?

Reply:
because event when you sign up using open id you still first sign in using open id and only after that you finalize sign up in gitea

Reply:
Actually I think it is a new problem/nit of the new UI design.

In old UI: there is a separate tab "OpenID", everything is done there, so no confusion.

In new UI: if EnableSignUp=false, but there is a "Sign in with OIDC" on the sign-up page, I think it is confusing to end users. IMO the sign-up page should only show what is really for "sign-up", but not something for other purposes.

Anyway, it isn't serious, so there is no objection from my side if people like it.